### DISCUSSION TITLE
Zigbee2MQTT can't connect at start service

### DISCUSSION THREAD

**Discussion by rollingafull** [tteck/Proxmox#910] [Help Needed] [answered]
hi,

After serveral installs when i start service with 'npm start' always receive same message, 'MQTT error: connect ECONNREFUSED 192.168.2.103:1883'
I try with default dhcp config and with fixed ip.
I try with default configuration.yml only changing device and with server mqtt:// with ipaddress and localhost.
Always same message.

Has anybody this issue?

"root@zigbee2mqtt:/opt/zigbee2mqtt# npm start

> zigbee2mqtt@1.29.0 start
> node index.js

Building Zigbee2MQTT... (initial build), finished
Zigbee2MQTT:info  2023-01-06 07:05:28: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-01-06.07-05-28' filename: log.txt
Zigbee2MQTT:info  2023-01-06 07:05:28: Starting Zigbee2MQTT version 1.29.0 (commit #dddce571)
Zigbee2MQTT:info  2023-01-06 07:05:28: Starting zigbee-herdsman (0.14.83)
Zigbee2MQTT:info  2023-01-06 07:05:44: zigbee-herdsman started (reset)
Zigbee2MQTT:info  2023-01-06 07:05:44: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20220219,"transportrev":2},"type":"zStack3x0"}'
Zigbee2MQTT:info  2023-01-06 07:05:44: Currently 0 devices are joined:
Zigbee2MQTT:info  2023-01-06 07:05:44: Zigbee: disabling joining new devices.
Zigbee2MQTT:info  2023-01-06 07:05:44: Connecting to MQTT server at mqtt://192.168.2.103
Zigbee2MQTT:error 2023-01-06 07:05:44: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:44: MQTT failed to connect, exiting...
Zigbee2MQTT:info  2023-01-06 07:05:44: Stopping zigbee-herdsman...
Zigbee2MQTT:error 2023-01-06 07:05:45: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:46: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:47: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:48: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:49: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:50: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:51: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:05:52: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:info  2023-01-06 07:05:52: Stopped zigbee-herdsman
root@zigbee2mqtt:/opt/zigbee2mqtt#"

My config file:

"homeassistant: false
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://192.168.2.103
serial:
  port: >-
    /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_4cbb26a42c3bec1186fca4957a0af07f-if00-port0
advanced:
  network_key:
    - 127
    - 137
    - 151
    - 69
    - 199
    - 25
    - 157
    - 39
    - 71
    - 160
    - 11
    - 92
    - 102
    - 205
    - 192
    - 59
  homeassistant_legacy_entity_attributes: false
  legacy_api: false
  legacy_availability_payload: false
frontend: true
device_options:
  legacy: false"

**Comment by tteck** [comment]
`'mqtt://192.168.2.103:1883'`

**Comment by rollingafull** [comment]
yes i also try it without sucess.

**Reply by tteck** [reply]
```
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.2.103:1883'
  user: usr
  password: pwd
  keepalive: 60
  reject_unauthorized: true
  version: 4
```

**Reply by tteck** [reply]
check indents

**Comment by rollingafull** [comment]
thanks fora quick reply.

Yes i also try your config previously without sucess.

"root@zigbee2mqtt:/opt/zigbee2mqtt# cat /opt/zigbee2mqtt/data/configuration.yaml
homeassistant: false
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.2.103:1883'
  user: rolling
  password: mypassword
  keepalive: 60
  reject_unauthorized: true
  version: 4
serial:
  port: >-
    /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_4cbb26a42c3bec1186fca4957a0af07f-if00-port0
advanced:
  network_key:
    - 127
    - 137
    - 151
    - 69
    - 199
    - 25
    - 157
    - 39
    - 71
    - 160
    - 11
    - 92
    - 102
    - 205
    - 192
    - 59
  homeassistant_legacy_entity_attributes: false
  legacy_api: false
  legacy_availability_payload: false
frontend: true
device_options:
  legacy: false
root@zigbee2mqtt:/opt/zigbee2mqtt# npm start

> zigbee2mqtt@1.29.0 start
> node index.js

Zigbee2MQTT:info  2023-01-06 07:29:02: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2023-01-06.07-29-02' filename: log.txt
Zigbee2MQTT:info  2023-01-06 07:29:02: Starting Zigbee2MQTT version 1.29.0 (commit #dddce571)
Zigbee2MQTT:info  2023-01-06 07:29:02: Starting zigbee-herdsman (0.14.83)
Zigbee2MQTT:info  2023-01-06 07:29:03: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2023-01-06 07:29:03: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20220219,"transportrev":2},"type":"zStack3x0"}'
Zigbee2MQTT:info  2023-01-06 07:29:03: Currently 0 devices are joined:
Zigbee2MQTT:info  2023-01-06 07:29:03: Zigbee: disabling joining new devices.
Zigbee2MQTT:info  2023-01-06 07:29:03: Connecting to MQTT server at mqtt://192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:03: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:03: MQTT failed to connect, exiting...
Zigbee2MQTT:info  2023-01-06 07:29:03: Stopping zigbee-herdsman...
Zigbee2MQTT:error 2023-01-06 07:29:04: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:05: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:06: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:07: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:08: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:09: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:10: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:error 2023-01-06 07:29:11: MQTT error: connect ECONNREFUSED 192.168.2.103:1883
Zigbee2MQTT:info  2023-01-06 07:29:11: Stopped zigbee-herdsman
root@zigbee2mqtt:/opt/zigbee2mqtt#
"

what is the user and password that i must put on configuration.yml?

**Comment by tteck** [comment, marked as answer]
```
frontend:
  port: 9442
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.2.103:1883'
  user: usr
  password: pwd
  keepalive: 60
  reject_unauthorized: true
  version: 4
serial:
  port: /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_4cbb26a42c3bec1186fca4957a0af07f-if00-port0
advanced:
  pan_id: GENERATE
  network_key: GENERATE
  channel: 20
```
Are you putting your MQTT user/password in?

**Reply by tteck** [reply]
The config above should work after you add your MQTT credentials

**Reply by rollingafull** [reply]
well this is the problem sure, but what is the user and password that i must put, i don't know it.
During installation process not ask for it? is there a default user/pass?

**Reply by tteck** [reply]
It's the first step in the instructions

**Reply by rollingafull** [reply]
sorry @tteck but i'm blind, i can't see this instructions... :-(

**Reply by tteck** [reply]
![Screenshot 2023-01-06 2 44 37 AM](https://user-images.githubusercontent.com/21342670/210954480-1bcc035f-9f6b-4f5b-b003-b23583d6ab16.png)

**Reply by rollingafull** [reply]
Thanks, i whill try it.

But i'm installing this container, and it not has this instructions, sorry.
Maybee is necesary update it?

![imagen](https://user-images.githubusercontent.com/62690523/210956068-7127441f-84e6-4b38-aa3e-46073b55307a.png)

**Reply by tteck** [reply]
You must have a _**working**_ MQTT broker _**before**_ installing Zigbee2**MQTT**

**Reply by rollingafull** [reply]
I now undertand it. Sorry.
Is necesary a MQTT LXC for connect, i not check this requisite.

Thanks for your help.
